### ISSUE TITLE
metadata command not working on Linux (Mono)

### ISSUE THREAD

**Issue by dlech** [dotnet/docfx#1708]
### Functional impact

`docfx metadata` is not usable on Linux.

### Minimal repro steps

* Ubuntu 16.04
* Mono from [Mono repository](http://www.mono-project.com/download/#download-lin-ubuntu)

        mono --version
        Mono JIT compiler version 5.0.1.1 (2017-02/5077205 Thu May 25 09:19:18 UTC 2017)
* [docfx v2.17.7](https://github.com/dotnet/docfx/releases/tag/v2.17.7)
* Clone [docfx-seed repo](https://github.com/docascode/docfx-seed)
* Run `mono /path/to/docfx.exe metadata`

### Expected result

Metadata should be created.

### Actual result

If fails to create any data...

```
[17-05-28 12:35:53.204]Info:Config file docfx.json found, start generating metadata...
Could not load signature of Microsoft.CodeAnalysis.SymbolSearch.SymbolSearchUpdateEngine+RemoteControlService:CreateClient due to: Could not load file or assembly 'Microsoft.V
isualStudio.RemoteControl, Version=14.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. assembly:Microsoft.VisualStudio.RemoteControl, Versi
on=14.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a type:<unknown type> member:<none>
Could not load signature of Microsoft.CodeAnalysis.SymbolSearch.IRemoteControlService:CreateClient due to: Could not load file or assembly 'Microsoft.VisualStudio.RemoteContro
l, Version=14.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. assembly:Microsoft.VisualStudio.RemoteControl, Version=14.0.0.0, Culture=neu
tral, PublicKeyToken=b03f5f7f11d50a3a type:<unknown type> member:<none>
[17-05-28 12:35:53.682]Warning:Error opening solution /home/david/work/junk/docfx-seed/src/SampleClass1/SampleClass1.sln: Could not resolve the signature of a virtual method.
Ignored.
[17-05-28 12:35:53.698]Warning:No metadata is generated for .
[17-05-28 12:35:53.729]Info:Completed executing in 393.7172 milliseconds.
```

### Further technical details

According to the error message, `Microsoft.VisualStudio.RemoteControl` is missing. This does not sound like something that would be included in Mono, so it either needs to be included in the docfx.zip or the dependency should be removed.

**Comment by dlech** [comment]
FYI, I have just tried DocFX on Windows 10 using the choco package and the seed repo with basically the same results (can't generate metadata), so it does not seem that this problem is restricted to just Linux/Mono.

```
PS C:\Users\david\Documents\Junk\docfx-seed> docfx metadata --loglevel Verbose
[17-06-06 08:03:31.112]Info:Config file docfx.json found, start generating metadata...
[17-06-06 08:03:33.573]Verbose:Successfully generated metadata C:/Users/david/Documents/Junk/docfx-seed/src/SampleClass1
/SomeVbClass/obj/xdoc/cache/obj for SomeVbClass
[17-06-06 08:03:33.657]Verbose:Successfully generated metadata C:/Users/david/Documents/Junk/docfx-seed/src/SampleClass1
/CatLibrary/obj/xdoc/cache/obj for CatLibrary
[17-06-06 08:03:33.674]Verbose:Successfully generated metadata C:/Users/david/Documents/Junk/docfx-seed/src/SampleClass1
/BaseClass/obj/xdoc/cache/obj for BaseClass
[17-06-06 08:03:33.689]Warning:No metadata is generated for SomeVbClass,CatLibrary,BaseClass.
[17-06-06 08:03:33.704]Info:Completed executing in 2326.677 milliseconds.


Build succeeded with warning.
[17-06-06 08:03:33.721]Warning:No metadata is generated for SomeVbClass,CatLibrary,BaseClass.
        1 Warning(s)
        0 Error(s)
```

**Comment by dlech** [comment]
On further inspection, on Windows, it seems to be a different issue. You have to manually specify the source code files. So, no problem on Windows other than the error message is not informative.

**Comment by dlech** [comment]
Now that I understand the separate issue I was seeing on Windows (with the same warning message), I was able to get things working on Linux/Mono. I just grabbed the .dll from <https://www.nuget.org/packages/Microsoft.VisualStudio.RemoteControl/> and threw it in with the others from the .zip file.

**Comment by vicancy** [comment]
Hi @dlech Are you able to generate documents under mono after the missing dll is added? In my local machine, `mono docfx` fails with with `no metadata generated`.

**Comment by dlech** [comment]
There are two separate issue here. The first is...

```
Could not load signature of Microsoft.CodeAnalysis.SymbolSearch.SymbolSearchUpdateEngine+RemoteControlService:CreateClient due to: Could not load file or assembly 'Microsoft.V
isualStudio.RemoteControl, Version=14.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. assembly:Microsoft.VisualStudio.RemoteControl, Versi
on=14.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a type:<unknown type> member:<none>
Could not load signature of Microsoft.CodeAnalysis.SymbolSearch.IRemoteControlService:CreateClient due to: Could not load file or assembly 'Microsoft.VisualStudio.RemoteContro
l, Version=14.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. assembly:Microsoft.VisualStudio.RemoteControl, Version=14.0.0.0, Culture=neu
tral, PublicKeyToken=b03f5f7f11d50a3a type:<unknown type> member:<none>
```

Adding the .dll fixes this issue.

The second is...

```
[17-05-28 12:35:53.698]Warning:No metadata is generated for .
```

This is fixed by editing `docfx.json` and adding `**/*.cs` to the source files.

```
{
  "metadata": [
    {
      "src": [
        {
          "files": [ "**/*.sln", "**/*.cs" ],  <-- FIX
          "exclude": [ "**/bin/**", "**/obj/**" ],
          "cwd": "src"
        }
      ],
...
```

I made PR #1734 to get a more useful error message for the second issue. The second issue seems to be caused by the lack of support for the new csproj file format.

**Comment by vicancy** [comment]
The error is fixed in v2.18.3

**Comment by dlech** [comment]
Confirmed both issues are fixed under Mono. Thanks!
